code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -422,14 +422,22 @@
 		if(!isSwitchingStates && state == 1)
 			Close()
 
+/obj/structure/mineral_door/resin/proc/close_blocked()
+	for(var/turf/turf in locs)
+		for(var/mob/living/living_mob in turf)
+			var/datum/component/weed_food/weed = living_mob.GetComponent(/datum/component/weed_food)
```

Comment:
You're not supposed to use GetComponent at all, it defeats the whole point of components, use a signal

Reply:
If so, then other uses of GetComponent should be refactored and the comments for GetComponent should reflect this. Regardless simple enough to convert over to the style of signal handling that is frowned upon.

Reply:
Well it's in the ECS documentation

![image](https://github.com/cmss13-devs/cmss13/assets/604624/5d1b827b-93c2-4a53-a726-d1cd1301ec9d)

You could use a new trait but leveraging the component sounds good to me, it's just the whole point of them is to delegate the behavior to it through signal/proc rather than acting upon it conditionally in the calling code